Sources/S3/Extensions/S3+Private.swift

```diff
@@ -1,28 +1,55 @@
-//
-//  S3+Private.swift
-//  S3
-//
-//  Created by Ondrej Rafaj on 19/04/2018.
-//
-
 import Foundation
 import Vapor
 import HTTP
 
-
 extension S3 {
     
     /// Make an S3 request
-    func make(request url: URL, method: HTTPMethod, headers: HTTPHeaders, data: Data? = nil, on container: Container) throws -> Future<Response> {
-        let client = try container.make(Client.self)
-        let request = Request(using: container)
-        request.http.method = method
-        request.http.headers = headers
+    func make(request url: URL, method: HTTPMethod, headers: HTTPHeaders, data: Data? = nil, cachePolicy: URLRequest.CachePolicy = .useProtocolCachePolicy, on container: Container) throws -> Future<Response> {
+        var request = URLRequest(url: url, cachePolicy: cachePolicy)
+        request.httpMethod = method.string
+        request.httpBody = data ?? Data()
```

Comment:
can we just skip setting body altogether so we don't have to create an empty object?